### ISSUE TITLE
Native Google Login not working on Release

### ISSUE THREAD

**Issue by Kokobe** [neevaco/neeva-android#1213]
On `debug` it works fine:

https://user-images.githubusercontent.com/20916043/213866886-f88f2f0e-8376-4e0d-b4ed-02130c13512a.mp4

But on `release`, this happens:

https://user-images.githubusercontent.com/20916043/213866966-85072cba-36b1-467a-894e-d13e13c6d33c.mp4

**Comment by Kokobe** [comment]
It looks like this is due to the `ActivityResult.resultCode` being equal to 0 when it doesn't work on release. Perhaps looking at the `resultCode` isn't a good way to determine user cancellation.

**Comment by Kokobe** [comment]
Ah but the root of the problem is that I'm getting a `Login API Failure` which causes the `resultCode` = 0 part...

**Comment by Kokobe** [comment]
@dan-alcantara-neeva Going to try to coordinate with Carl to fix htis

**Comment by Kokobe** [comment]
Nevermind! It works when you download the build from the Play Store :) 


https://user-images.githubusercontent.com/20916043/213869218-162ad726-ade4-495b-afc3-e17e7350e0a0.mp4

**Comment by Kokobe** [comment]
Turns out it will fail for some accounts:


https://user-images.githubusercontent.com/20916043/213872912-810d1277-c646-4df4-bcd9-521b8a6b233e.mp4

**Comment by Kokobe** [comment]
Opening the uri that Google Sign In returns to me in a SingleTabActivity returns me an intent that gives the following error message: "This email is associated with an existing Neeva account".

https://user-images.githubusercontent.com/20916043/213873638-8c1e6da2-1aad-4829-89ea-7fa73fe3bf93.mp4


Even though when I use a SingleTabActivity, it works without any problem:

https://user-images.githubusercontent.com/20916043/213873664-deea1676-dde6-47d5-95f3-23a4607cb537.mp4

**Comment by Kokobe** [comment]
@clischeske It seems that when I try to login using Google Native Sign In on release (see videos above) for my account jehanc@uci.edu, 
it returns me a URI with an error saying "This email is associated with an existing Neeva account", but sign-in with Google on a Tab it will log me in without any problems. 

I don't think this is release blocking because I have a fallback in the release code to use the SingleTabActivity as a fallback if the Google Native Sign in does not work (see [here](https://paper.dropbox.com/doc/TODO-Android-Premium--BxNN35~7283NtrFvTgiMR7isAg-iiThY2l0WQvTbvbhsy8Nw))... but it would be really good to look into

**Comment by clischeske** [comment]
Yeah there are still a couple of kinks that I need to work out of Google Native sign-in. It's not ready for launch yet.

**Comment by Kokobe** [comment]
Confirmed that Google Native Sign In (or rather "sign up") works with accounts with no prior Neeva account. 

For accounts that did have a Google account with Neeva, (my jehanc@uci.edu account for example), it will not work. 

For accounts that signed up with OKTA who try to login with Native Google Login, it will not work (which is expected and OK)

**Comment by Kokobe** [comment]
If I use a `@gmail` account and sign up on the Neeva website, Native Google Login will not work for me either. So there doesn't seem to be a distinction between `@uci.edu` and `@gmail.com` in this error case
